File: Java_Herbert Schild/Selction Sort/src/test1/SelectionSort.java
package test1;

import java.util.Scanner;

public class SelectionSort
{
	private int elementNumber;
	private int[] arr;
	private int min;
	private int temp;
	
	Scanner sc = new Scanner(System.in);
	
	/**
	 * It will enter the number of element
	 */
	void inputElementNumber()
	{
		System.out.println("\nEnter number of element for sorting :\n");
		elementNumber = sc.nextInt();
	}
	
	/**
	 * It will create the array for the entered element number
	 */
	void arrArrayCreation()
	{
		arr = new int[elementNumber];
	}
	
	/**
	 * It will take input and store in the array
	 */
	void arrArrayInput()
	{
		System.out.println("\n\nEnter " + elementNumber + " numbers :\n");
		for(int i=0; i<elementNumber; i++)
		{
			arr[i] = sc.nextInt();
		}
	}
	
	/**
	 * It will show the inputed array before sorting
	 */
	void displayBeforeSorting()
	{
		System.out.println("\n\nArray before sorting using SELCTION SORT :\n");
		for(int i=0; i<elementNumber; i++)
		{
			System.out.println("arr[" + i + "] ---> " + arr[i]);
		}
	}
	
	/**
	 * It will sort the numbers using SELCTION SORT
	 */
	void selectionSort()
	{
		for(int i=0; i<elementNumber-1; i++)
		{
			//find the index of the smallest number
			min = i;
			for(int j=i+1; j<elementNumber; j++)
			{
				if(arr[min] > arr[j])
				{
					min=j;
				}
			}
			
			if(i!=min)
			{
				temp = arr[i];
				arr[i] = arr[min];
				arr[min] = temp;
			}
			
			/**
			 * It will print the array after every pass
			 */
			System.out.println("\n\nArray after pass " + (i+1) + " :\n");
			for(int k=0; k<elementNumber; k++)
			{
				System.out.println("arr[" + k + "] ---> " + arr[k]);
			}
		}
	}
	
	/**
	 * It will show the array after sorting using SELCTION SORT
	 */
	void displayAfterSorting()
	{
		System.out.println("\n\nArray after sorting using SELCTION SORT :\n");
		for(int i=0; i<elementNumber; i++)
		{
			System.out.println("arr[" + i + "] ---> " + arr[i]);
		}
	}
}

File: C++ Program/SelectionSort.cpp
/*		WRITE A PROGRAM OF SORTING USING SELECTION SORT		*/

#include<stdio.h>
#include<conio.h>
#define MAX 100

int main()
{
	int array[MAX],i,j,n,temp,min;
	
	/*It will take the number of or size of array as user input*/
	printf("\nEnter the numbers of element :\n");
	scanf("%d", &n);
	printf("\nThe array size is :  %d",n);
	
	/*It will take the elements of an array form user*/
	printf("\n\nEnter the values of the array to be sorted with SELECTION SORT :\n");
	for(i=0; i<n; i++)
	{
		printf("\n\tArray[%d] = ",i);
		scanf("%d", &array[i]);
	}
	
	/*It will show the array before sorting*/
	printf("\n\nArray before sorting :\n");
	printf("==========================\n");
	for(i=0; i<n; i++)
	{
		printf("\n\tArray[%d] = %d",i,array[i]);
	}
	
	/*It is the two basic for loop for SELECTION SORT*/
	for(i=0; i<n-1; i++)
	{
		/*Now it will find the index of smallest number*/
		min = i;
		for(j=i+1; j<n; j++)
		{
			if(array[min] > array[j])
			{
				min = j;
			}
		}
		
		/*To avoid swapping of an element with itself*/
		if(i != min)
		{
			temp = array[i];
			array[i] = array[min];
			array[min] = temp;
		}
	}
	
	/*It will show the array after SELECTION sorting*/
	printf("\n\nArray after SELECTION sorting :\n");
	printf("===================================\n");
	for(i=0; i<n; i++)
	{
		printf("\n\tArray[%d] = %d",i,array[i]);
	}
	
	getch();
	return 0;
}